### PR TITLE
Add featured suggestions to the vertical question screen

### PR DESCRIPTION
**Pull request by taipeicoder**
#### Changes proposed in this Pull Request

This PR adds featured vertical suggestions to the vertical questions screen. 
Featured vertical suggestions will be shown to users when they initially click on the empty vertical question input field without having typed anything yet. Note that the list of featured vertical suggestions is not finalized and is very likely to be updated.

The idea behind displaying featured suggestions is two-fold:
1. Showing what the input field is for, and what they can expect from it.
2. Showing the most popular verticals chosen by users (or perhaps the most "verticalization-ready" ones).

Screen Shot 2022-04-19 at 10 43 35 AM

#### Testing instructions
1. Head to the vertical question screen `/setup/vertical?flags=signup/site-vertical-step&siteSlug={siteSlug}`
2. Click on the input text.
3. Expect to see the list of featured verticals in the suggestion dropdown.
4. Type something.
5. Expect to see the list of verticals refined to what you typed.
6. Clear what you wrote.
7. Expect to see the list of featured verticals shown again.

Related to #61526

### CODE REVIEW

File: client/components/select-vertical/suggestion-search.tsx [hunk 1]
```diff
@@ -21,35 +23,71 @@ const SelectVerticalSuggestionSearch: FC< Props > = ( {
...
+			setIsShowSuggestions( true );
 			onInputChange?.( event.target.value );
 		},
-		[ setIsShowSuggestions, onInputChange ]
+		[ setIsShowSuggestions ]
+	);
+
+	const handleToggleSuggestionsBlur = useCallback(
+		( event ) => {
+			if ( event && event.relatedTarget?.contains( inputRef.current ) ) {
```
Comment (arthur791004) [inline client/components/select-vertical/suggestion-search.tsx:72]: What does this check want to do?

File: client/landing/stepper/declarative-flow/internals/steps-repository/site-vertical/index.tsx [hunk 2]
```diff
@@ -33,8 +33,13 @@ const SiteVertical: Step = function SiteVertical( { navigation } ) {
...
 		event.preventDefault();
 
-		if ( site && vertical ) {
-			const { value, label } = vertical;
+		if ( site ) {
+			let value = '';
+			let label = '';
+			if ( vertical ) {
+				value = vertical.value;
+				label = vertical.label;
```
Comment (arthur791004) [inline client/landing/stepper/declarative-flow/internals/steps-repository/site-vertical/index.tsx]: When I go back to the vertical step, clean up the selection and then click the “Continue” button, I found the saved vertical is the same as the previous one.

File: client/components/select-vertical/index.tsx [hunk 3]
```diff
@@ -57,6 +64,7 @@ const SelectVertical: React.FC< Props > = ( { defaultVertical, isSkipSynonyms, o
 				suggestions={
 					! isDebouncing ? mapManySiteVerticalsResponseToVertical( suggestions || [] ) : []
 				}
+				featuredSuggestions={ mapManySiteVerticalsResponseToVertical( featured || [] ) }
```
Comment (arthur791004) [inline client/components/select-vertical/index.tsx]: Could we just use `suggestions`? That being said, if the `searchTerm` is empty, we use the feature suggestions

**Review by arthur791004 — COMMENTED** [review]
Looks good and here are minor notes:
1. Could we also update the color and font-weight of the “SUGGESTIONS” text? I found it's not aligned with the figma.
2. I found when we select the vertical, the component will still send API to query that term. Could we avoid this?

File: client/components/select-vertical/index.tsx (hunk 3 above)
Comment (taipeicoder) [reply client/components/select-vertical/index.tsx]: Sounds good to me 🙂

File: client/landing/stepper/declarative-flow/internals/steps-repository/site-vertical/index.tsx (hunk 2 above)
Comment (taipeicoder) [reply client/landing/stepper/declarative-flow/internals/steps-repository/site-vertical/index.tsx]: Good catch!

File: client/components/select-vertical/suggestion-search.tsx (hunk 1 above)
Comment (taipeicoder) [reply client/components/select-vertical/suggestion-search.tsx:72]: This check is to keep the suggestion dropdown open when the focus moves between the input field and the toggle icon (button). I've added comments to explain the intention.

**Comment by taipeicoder** [comment]
> Looks good and here are minor notes:
> 
> 1. Could we also update the color and font-weight of the “SUGGESTIONS” text? I found it's not aligned with the figma.
> 2. I found when we select the vertical, the component will still send API to query that term. Could we avoid this?

1. Good catch, I've updated it!.
2. Do you mean when there is a previously saved vertical? In that case we need to send an API since we only store the vertical ID, so we'll need to hit the API to retrieve the vertical name. If that's not what you're referring, could you provide more details? Thanks!

**Comment by arthur791004** [comment]
> Do you mean when there is a previously saved vertical? In that case we need to send an API since we only store the vertical ID, so we'll need to hit the API to retrieve the vertical name. If that's not what you're referring, could you provide more details? Thanks!

https://user-images.githubusercontent.com/13596067/164198387-73a45168-d8be-4c96-9d1d-cd3b54b8c564.mov

You could see the above video. After I select the suggestion value “Internet Cafes”, it still uses it as a term to query the suggestions. However, this vertical comes from the suggestion list and I think we have enough information (ex: id) and we don't need to query again.

The reason to trigger the API is we also set the term when the user selects a vertical. Then, the API is triggered as the search term is changed.
https://github.com/Automattic/wp-calypso/blob/e5b912b18aad57869f9d41069a00ed63584a6e39/client/components/select-vertical/suggestion-search.tsx#L119

**Comment by taipeicoder** [comment]
> > Do you mean when there is a previously saved vertical? In that case we need to send an API since we only store the vertical ID, so we'll need to hit the API to retrieve the vertical name. If that's not what you're referring, could you provide more details? Thanks!
> 
>  vertical-on-select.mov 
> You could see the above video. After I select the suggestion value “Internet Cafes”, it still uses it as a term to query the suggestions. However, this vertical comes from the suggestion list and I think we have enough information (ex: id) and we don't need to query again.
> 
> The reason to trigger the API is we also set the term when the user selects a vertical. Then, the API is triggered as the search term is changed.
> 
> https://github.com/Automattic/wp-calypso/blob/e5b912b18aad57869f9d41069a00ed63584a6e39/client/components/select-vertical/suggestion-search.tsx#L119

While the current approach "wastes" an API call since we already know about the vertical, but the idea behind calling `onInputChange` is because we're replacing the search query with the vertical name users selected. 

For instance, if the user selects "Food", the component updates the suggestion list based on the term "Food", which would be different with the previous list of suggestions. In this case, the API query is needed. 

Before selecting "Food":
Screen Shot 2022-04-21 at 10 57 38 AM

After selecting "Food":
Screen Shot 2022-04-21 at 10 57 46 AM

If we avoid calling the API when users select a vertical, then when the user clicks again the input field or clicks on the toggle icon, we will have to call the API again to obtain the new list (the second screenshot). In this case there will be a loading state since the list hasn't been retrieved yet. What are your thoughts on it?

**Comment by arthur791004** [comment]
Okay, that makes sense!

**Review by arthur791004 — APPROVED** [review]

**Pull request merged**